### PR TITLE
RLM-1208 Ensure ssh returns failures on script exit

### PR DESCRIPTION
**Pull request by antonym**
Existing method wasn't returning failure exit codes back
to originating script.  Refactoring to ensure a
failure is returned if infra1 deploy fails during the
process.

Issue: [RLM-1208](https://rpc-openstack.atlassian.net/browse/RLM-1208)

### CODE REVIEW

**Comment by antonym** [comment]
Testing here:

https://rpc.jenkins.cit.rackspace.net/view/RPC-O%20Artifacts/job/PM_rpc-openstack-master-xenial_mnaio_loose_artifacts-swift-deploy/

**Comment by antonym** [comment]
Exits properly now if deploy fails.

**Comment by antonym** [comment]
recheck_cit_artifact

**Comment by odyssey4me** [comment]
Note that this will need to be ported back into the master-rc branch to ensure that the test there works.

File: gating/pre_merge_test/run_deploy_mnaio.sh [hunk 1]
```diff
@@ -127,7 +127,32 @@ elif [[ "${RE_JOB_SCENARIO}" == "ironic" ]]; then
...
+popd
+pushd /opt/openstack-ansible/scripts
+  python pw-token-gen.py --file /etc/openstack_deploy/user_secrets.yml
+popd
+pushd /opt/openstack-ansible/playbooks
+  openstack-ansible setup-everything.yml
+popd
+EOF
+
+# sync rpc-o to infra prepare rpc-o configs and
```
Comment (rexredinger) [inline gating/pre_merge_test/run_deploy_mnaio.sh]: Was something supposed to be after the "and"?

Comment (antonym) [reply gating/pre_merge_test/run_deploy_mnaio.sh]: Yeah, good catch, just needed to be moved back a bit.

**Comment by antonym** [comment]
@odyssey4me yeah, I popped https://rpc-openstack.atlassian.net/browse/RLM-1292 to track addition the additional things to the MNAIO job.  Wanted to get that stable first before we lump more stuff on it.  I'll also pop a PR to backport to master-rc.

**Pull request merged**